# Patch: Force Megatron-LM to use system packages.
#
# Problem:
#   UV really doesn't want to use the container dependecies when creating a venv.
#
# What this patch does:
#   We force uv to not install any of the already available libraries.
#
# Where in the code:
#   File: pyproject.toml
#   Location: [tool.uv.sources] toml section.
#
# How to fix if this patch breaks on a future Megatron version:
#   Find the new location of [tool.uv] and update the patch.
#
# Applied to: Megatron-LM core_v0.17.0
#
diff --git a/pyproject.toml b/pyproject.toml
index 9fd20d70e..99c278131 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -195,6 +195,25 @@ override-dependencies = [
     "torch; sys_platform == 'never'",
     "torchvision; sys_platform == 'never'",
     "triton; sys_platform == 'never'",
+    "transformer-engine; sys_platform == 'never'",
+    "nvidia-resiliency-ext; sys_platform == 'never'",
+    "cuda-bindings; sys_platform == 'never'",
+    "cuda-pathfinder; sys_platform == 'never'",
+    "cuda-python; sys_platform == 'never'",
+    "nvidia-cudnn-frontend; sys_platform == 'never'",
+    "nvidia-cutlass-dsl; sys_platform == 'never'",
+    "nvidia-ml-py; sys_platform == 'never'",
+    "nvidia-modelopt; sys_platform == 'never'",
+    "nvtx; sys_platform == 'never'",
+    "numpy; sys_platform == 'never'",
+    "nvidia-resiliency-ext; sys_platform == 'never'",
+    "scipy; sys_platform == 'never'",
+    "pandas; sys_platform == 'never'",
+    "pyarrow; sys_platform == 'never'",
+    "protobuf; sys_platform == 'never'",
+    "onnx; sys_platform == 'never'",
+    "onnx-ir; sys_platform == 'never'",
+    "onnxscript; sys_platform == 'never'",
 ]
 
 [tool.uv.sources]
